types.go

```diff
@@ -15,33 +15,39 @@ import (
 
 const defaultGradientAbsTol = 1e-6
 
-// EvaluationType is used by the optimizer to specify information needed
-// from the objective function.
-type EvaluationType int
+// EvaluationType is used by the optimizer to specify information needed from
+// the objective function. The types can be composed together using the binary
+// or operator.
```

Comment:
Our users probably haven't used binary or much, so "using the binary or operator, for example 'FunctionEvaluation | GradEvaluation' to evaluate both the function value and the gradient"